api/registrations/views.py

```diff
@@ -0,0 +1,174 @@
+
+        query = base_query & permission_query
+        return query
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        query = self.get_query_from_request()
+        return Node.find(query)
+
+
+class RegistrationDetail(generics.RetrieveAPIView, RegistrationMixin):
+    """Node Registrations.
+
+    Registrations are read-only snapshots of a project. This view shows details about the given registration.
+
+    Each resource contains the full representation of the registration, meaning additional requests to an individual
+    registrations's detail view are not necessary.
+
+    ##Registration Attributes
+
+    Registrations have the "registrations" `type`.
+
+        name               type               description
+        ---------------------------------------------------------------------------------
+        title              string             title of the registered project or component
+        description        string             description of the registered node
+        category           string             node category, must be one of the allowed values
+        date_created       iso8601 timestamp  timestamp that the node was created
+        date_modified      iso8601 timestamp  timestamp when the node was last updated
+        tags               array of strings   list of tags that describe the registered node
+        fork               boolean            is this project a fork?
+        registration       boolean            has this project been registered?
+        collection         boolean            is this registered node a collection of other nodes?
+        dashboard          boolean            is this registered node visible on the user dashboard?
+        public             boolean            has this registration been made publicly-visible?
+        retracted          boolean            has this registration been retracted?
+        date_registered    iso8601 timestamp  timestamp that the registration was created
+
+    ##Relationships
+
+    ###Registered from
+
+    The registration is branched from this node.
+
+    ###Registered by
+
+    The registration was initiated by this user.
+
+    ##Links
+
+        self:  the canonical api endpoint of this registration
+        html:  this registration's page on the OSF website
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublic,
+        ReadOnlyIfRegistration,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NODE_BASE_WRITE]
+
+    serializer_class = RegistrationDetailSerializer
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        registration = self.get_node()
+        if not registration.is_registration:
+            raise NotFound('This is not a registration.')
```

Comment:
Not sure whether 400 or 404 is the better error code here: something *was* found, after all.